### ISSUE TITLE
does not handle * as type in generics

### ISSUE THREAD

**Issue by benkuly** [mockative/mockative#29]
An interface with a function
```
suspend fun getEvent(context: MatrixEndpointContext<GetEvent, Unit, Event<*>>): Event<*> 
```
gets compiled to a mock with 
```
override suspend fun getEvent(context: MatrixEndpointContext<GetEvent, Unit, Event<EventContent>>): Event<EventContent> = io.mockative.Mockable.suspend<Event<EventContent>>(this, io.mockative.Invocation.Function("getEvent", listOf<Any?>(context)), false)
```
Running the test throws `Class ... is not abstract and does not implement abstract member public abstract suspend fun getEvent`. I think it's because mockative makes * to EventContent.

**Comment by Nillerr** [comment]
Hi @benkuly 

Thanks for reporting this issue, and apologies for not getting back to you sooner.

I have just released version `1.2.0` of Mockative, which features a complete rewrite of the code generation using [KotlinPoet](https://square.github.io/kotlinpoet/). I tested your case in this new version and I could not replicate it. Please try version `1.2.0` and reopen the issue if it remains 🙂

**Comment by Nillerr** [comment]
Hi again

I discovered an issue in `1.2.0` causing the generated code to be invalid. I have resolved this issue in `1.2.1`, which is on its way out.
